test/services/SelfEmploymentServiceSpec.scala

```diff
@@ -178,39 +188,94 @@ class SelfEmploymentServiceSpec extends SpecBase with MockitoSugar with Argument
+  "submitGatewayQuestionAndClearDependentAnswers" - {
     "return UserAnswers with cleared dependent pages when selected No" in {
-      val existingAllAnswers = SetAnswer
-        .setMany(businessId, emptyUserAnswers)(
-          SetAnswer(ZegvAllowancePage, true),
-          SetAnswer(ZegvClaimAmountPage, BigDecimal(200)),
-          SetAnswer(ZegvHowMuchDoYouWantToClaimPage, ZegvHowMuchDoYouWantToClaim.LowerAmount),
-          SetAnswer(ZegvOnlyForSelfEmploymentPage, true),
-          SetAnswer(ZegvTotalCostOfVehiclePage, BigDecimal(100)),
-          SetAnswer(ZegvUseOutsideSEPage, ZegvUseOutsideSE.Ten),
-          SetAnswer(ZegvUseOutsideSEPercentagePage, 300)
-        )
-        .success
-        .value
-
       val updatedAnswers =
         service
-          .submitGatewayQuestionAndClearDependentAnswers(ZeroEmissionGoodsVehiclePage, businessId, existingAllAnswers, newAnswer = false)
+          .submitGatewayQuestionAndClearDependentAnswers(ZeroEmissionGoodsVehiclePage, businessId, existingZegvAnswers, newAnswer = false)
           .futureValue
 
-      val expectedAnswers = emptyUserAnswers
-        .set(
-          ZeroEmissionGoodsVehiclePage,
-          false,
-          Some(businessId)
-        )
-        .success
-        .value
-        .data
+      val dbAnswers = repository.state(userAnswersId).data
+      assert(dbAnswers === expectedClearedAnswers)
+
+      assert(updatedAnswers.data === expectedClearedAnswers)
+    }
+  }
+
+  "submitGatewayQuestionAndRedirect" - {
+    "return a Redirect to the next page and cleared dependent pages when answer is 'No'" in {
+      val result =
+        service
+          .submitGatewayQuestionAndRedirect(ZeroEmissionGoodsVehiclePage, businessId, existingZegvAnswers, newAnswer = false, taxYear, NormalMode)
+          .futureValue
 
-      assert(updatedAnswers.data === expectedAnswers)
       val dbAnswers = repository.state(userAnswersId).data
+      assert(dbAnswers === expectedClearedAnswers)
+
+      assert(result.header.status === SEE_OTHER)
+      assert(result.header.headers.head._2 === ZeroEmissionGoodsVehiclePage.cyaPage(taxYear, businessId).url)
+    }
+  }
+
+  "persistAnswerAndRedirect" - {
+    "save answer to session repository and return a Redirect to the next page" in {
+      val result =
+        service
+          .persistAnswerAndRedirect(
+            ZeroEmissionGoodsVehiclePage,
+            businessId,
+            fakeDataRequest(existingZegvAnswers),
+            value = false,
+            taxYear,
+            NormalMode)
+          .futureValue
+
+      val expectedAnswers = existingZegvAnswers.set(ZeroEmissionGoodsVehiclePage, false, businessId.some).success.value
+      val dbAnswers       = repository.state(userAnswersId)
       assert(dbAnswers === expectedAnswers)
+
+      assert(result.header.status === SEE_OTHER)
+      assert(result.header.headers.head._2 === ZeroEmissionGoodsVehiclePage.cyaPage(taxYear, businessId).url)
```

Comment:
We could use Play's built in helpers for this, where we pass in in the future:
```scala
status(result) shouldBe ...
redirectLocation(result) shouldBe ...
```

Reply:
agreed, thanks